pkg/controllermanager/controller/shoot/shoot_maintenance_control.go

```diff
@@ -158,15 +158,15 @@ func (c *defaultMaintenanceControl) Maintain(shootObj *gardenv1beta1.Shoot, key
 		return nil
 	}
 
-	updateImage, machineImage, err := MaintainMachineImage(operation.Shoot.Info, operation.Shoot.CloudProfile, operation.Shoot.GetMachineImage())
+	machineImages, err := MaintainMachineImages(operation.Shoot.Info, operation.Shoot.CloudProfile, operation.Shoot.GetMachineImages())
 	if err != nil {
 		// continue execution to allow the kubernetes version update
 		handleError(fmt.Sprintf("Could not maintain machine image version: %s", err.Error()))
 	}
 
-	var updateMachineImage func(s *gardenv1beta1.Cloud)
-	if updateImage {
-		updateMachineImage = helper.UpdateMachineImage(operation.Shoot.CloudProvider, machineImage)
+	var updateMachineImages func(s *gardenv1beta1.Cloud)
+	if len(machineImages) > 0 {
+		updateMachineImages = helper.UpdateMachineImages(operation.Shoot.CloudProvider, machineImages)
```

Comment:
This only updates the machine images in the worker section. The default machine image in `.spec.cloud.<provider>.machineImage` is not maintained anymore. This needs to be changed back.